### ISSUE TITLE
Can I publish hybrid app (iOS) on AppCenter

### ISSUE THREAD

**Issue by AmirImam** [xamarin/MobileBlazorBindings#281]
Hi.
Is it possible now to publish hybrid app on AppCenter for testing ?
I tried it but it throws error
##[error]Xamarin.iOS task failed with error Error: /Library/Frameworks/Mono.framework/Versions/6_12_4/bin/msbuild failed with return code: 1. For guidance on setting up the build definition, see https://go.microsoft.com/fwlink/?LinkId=760847.

I even tried to do it with just default hybrid app template, didn't add any other code or libraries but throws same error.
And if is it not possible , is there any way to do it because I need it.
Thank you

**Comment by Eilon** [comment]
Hi @AmirImam I've never published on AppCenter but it should be exactly the same as any Xamarin-based app (which includes Mobile Blazor Bindings native and hybrid apps).

Have you been able to publish a regular Xamarin-based app to AppCenter? I recommend trying that first to rule out certain problems related specifically to Mobile Blazor Bindings.

**Comment by AmirImam** [comment]
> Hi @AmirImam I've never published on AppCenter but it should be exactly the same as any Xamarin-based app (which includes Mobile Blazor Bindings native and hybrid apps).
> 
> Have you been able to publish a regular Xamarin-based app to AppCenter? I recommend trying that first to rule out certain problems related specifically to Mobile Blazor Bindings.

Yes I have published Xamarin forms app to AppCenter for testing, and distributed iOS app for some real devices.
If there is nothing in blazor hybrid that prevents publishing, then maybe the problem in my publishing steps.

**Comment by Eilon** [comment]
Did you try the same publish steps in both apps? It's certainly possible that there are some Mobile Blazor Bindings specific steps required, but for the most part as far as the platform/publish is concerned, it's "just a regular Xamarin app."

**Comment by AmirImam** [comment]
Yes I tried same steps, and I'll try it again to make sure about it. But if there are MBB specific steps then I don't know them, so I thought that I can find them here

**Comment by Eilon** [comment]
Does the raw log on Azure DevOps show any other errors that might give insight on this? Unfortunately the error shown above doesn't say much.

**Comment by AmirImam** [comment]
No, only that error. I think the only available solution now is trying to publish again maybe I missing something

**Comment by Eilon** [comment]
If you can provide any additional info that would be very helpful. There could be an issue here in MBB but I probably won't have time to look at it very soon.

**Comment by AmirImam** [comment]
```
_CoreCompileColladaAssets:
Skipping target "_CoreCompileColladaAssets" because it has no inputs.
_BeforeCoreCompileSceneKitAssets:
Skipping target "_BeforeCoreCompileSceneKitAssets" because it has no inputs.
_BeforeCoreCompileSceneKitAssets:
Skipping target "_BeforeCoreCompileSceneKitAssets" because it has no inputs.
_BeforeCoreCompileSceneKitAssets:
Skipping target "_BeforeCoreCompileSceneKitAssets" because it has no inputs.
_CoreCompileSceneKitAssets:
Skipping target "_CoreCompileSceneKitAssets" because it has no inputs.
_BeforeCompileTextureAtlases:
Skipping target "_BeforeCompileTextureAtlases" because it has no inputs.
_BeforeCompileTextureAtlases:
Skipping target "_BeforeCompileTextureAtlases" because it has no inputs.
_BeforeCompileTextureAtlases:
Skipping target "_BeforeCompileTextureAtlases" because it has no inputs.
_CoreCompileTextureAtlases:
Skipping target "_CoreCompileTextureAtlases" because it has no inputs.
_BeforeCompileCoreMLModels:
  Directory "obj/iPhone/Debug/coremlc" doesn't exist. Skipping.
_CoreOptimizePngImages:
Skipping target "_CoreOptimizePngImages" because it has no outputs.
_CoreOptimizePropertyLists:
Skipping target "_CoreOptimizePropertyLists" because it has no inputs.
_CoreOptimizeLocalizationFiles:
Skipping target "_CoreOptimizeLocalizationFiles" because it has no inputs.
Done Building Project "/Users/runner/work/1/s/MyProject.V2.Mobile/MyProject.V2.Mobile.iOS/MyProject.V2.Mobile.iOS.csproj" (Rebuild target(s)) -- FAILED.
Done Building Project "/Users/runner/work/1/s/MyProject.V2.sln" (Rebuild target(s)) -- FAILED.

Build FAILED.
```
I found this in the AppCenter build log. Maybe it will be useful

**Comment by AmirImam** [comment]
Found these two

```
The application to execute does not exist: '/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk.Razor/build/netstandard2.0/../../tools/netcoreapp3.0/rzc.dll'
/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Component.targets(106,5): error : rzc generate exited with code 129. [/Users/runner/work/1/s/MarketTree.V2.Mobile/MarketTree.V2.Mobile/MarketTree.V2.Mobile.csproj]
```

**Comment by AmirImam** [comment]
Last thing, I could publish my MMB hybrid iOS app on TestFlight successfully. So now the problem is with AppCenter

**Comment by Eilon** [comment]
Is there any info here that helps? https://docs.microsoft.com/en-us/appcenter/build/troubleshooting/xamarin

There could be some limitation on App Center where the Razor SDK is somehow not available when building a Xamarin.Forms iOS app. Do you have the ability to specify which SDKs/dependencies your app requires on App Center?

**Comment by Eilon** [comment]
They do have some free support contact it seems as well: https://docs.microsoft.com/en-us/appcenter/help

I looked at the list of Supported Frameworks in https://docs.microsoft.com/en-us/appcenter/build/xamarin/ and I didn't see any mention of web/ASP.NET/Razor/Blazor so maybe there's a limitation on App Center's build system regarding Razor. That support link might help find some info.

I wish I could help more with this but I haven't tried App Center at all. I'm a bit more familiar with Azure Dev Ops but only for building libraries (such as Mobile Blazor Bindings) and not as much for building apps.
